src/components/ArticleList.jsx

```diff
@@ -1,16 +1,39 @@
 import { useEffect, useState } from "react";
-import { getAllArticles } from "../api/api";
+import { getTopics, getAllArticles } from "../api/api";
 import { Link, useParams } from "react-router-dom";
 import ArticleHeader from "./ArticleHeader";
 
 function ArticleList() {
   const [isError, setIsError] = useState(false);
+  const [urlIsWrong, setUrlIsWrong] = useState(false);
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const { topic } = useParams();
   const [sort_by, setSort_by] = useState("created_at");
   const [order, setOrder] = useState("DESC");
 
+  let { topic } = useParams();
+
+  useEffect(() => {
+    if (topic) {
+      getTopics()
+        .then((response) => {
+          const ValidTopics = response.data.topics;
+          if (
+            !ValidTopics.map((topicObj) => {
+              return topicObj.slug;
+            }).includes(topic)
+          ) {
+            setUrlIsWrong(true);
+            topic = "";
+          }
```

Comment:
You should already have this logic in your backend that checks the provided topic is valid for a query, we should be able to rely on this rather than do another check in the front end 🔍